hack/remove-old-versions.sh

```diff
@@ -20,7 +20,7 @@ VERSIONS_YAML_FILE=${VERSIONS_YAML_FILE:-"versions.yaml"}
 VERSIONS_YAML_DIR=${VERSIONS_YAML_DIR:-"pkg/istioversion"}
 
 function removeOldVersions() {
-    versions=$(yq eval '.versions[].name' "${VERSIONS_YAML_DIR}/${VERSIONS_YAML_FILE}" | tr $'\n' ' ')
+    versions=$(yq eval '.versions[] | select(.ref == null) | select(.eol != true) | .name' "${VERSIONS_YAML_DIR}/${VERSIONS_YAML_FILE}" | tr $'\n' ' ')
```

Comment:
Should we remove the whole file. It's not going to be used anymore.

Reply:
I mean `hack/remove-old-versions.sh` is not needed anymore as we will only use `eol` flag now.

Reply:
it's still needed, e.g. when branching off of `main`, and to delete `eol: true` charts

Reply:
ok, thanks. I will try it with 1.26 now.

Reply:
Sorry, I had bad assumption. I checked how is this script used and now I understand.